src/dmd/lexer.d

```diff
@@ -1880,7 +1880,7 @@ class Lexer : ErrorHandler
                     anyHexDigitsNoSingleUS = true;
                 if (base == 2 && !err)
                 {
-                    error("binary digit expected");
+                    error("radix %d digit expected, not `%c`", base, c);
```

Comment:
The base is always 2, it doesn't need to be an argument.

Reply:
I don't see anything wrong with `binary` either, instead of `radix 2`.

Reply:
The issue originally is that 0b5 prints "binary digit expected" while 0b8 prints "radix 2 digit expected not 8". It's the inconsistency that is being fixed.

But like I said, I think the whole thing can be fixed to be completely consistent and future proof. When #8451 is merged, I'll set up a new PR.